dxda.go

```diff
@@ -151,19 +151,27 @@ func makeRequestWithHeadersFail(requestType string, url string, headers map[stri
 		}
 	}
 
-	req, err := retryablehttp.NewRequest(requestType, url, bytes.NewReader(data))
-	check(err)
-	for header, value := range headers {
-		req.Header.Set(header, value)
-	}
-	resp, err := client.Do(req)
-	check(err)
-	defer resp.Body.Close()
-	status = resp.Status
-	body, _ = ioutil.ReadAll(resp.Body)
+	// Perpetually retry on 503 (e.g. platform downtime/throttling)
+	for {
+		req, err := retryablehttp.NewRequest(requestType, url, bytes.NewReader(data))
+		check(err)
+		for header, value := range headers {
+			req.Header.Set(header, value)
+		}
+		resp, err := client.Do(req)
+		check(err)
+		defer resp.Body.Close()
+		status = resp.Status
+		if status == "503" {
+			time.Sleep(5 * time.Second)
```

Comment:
Perhaps you want to take a look at how the dx-toolkit handles the backoff: https://github.com/dnanexus/dx-toolkit/blob/master/src/python/dxpy/__init__.py#L361

Reply:
Thanks!  For the initial releases of dxda, I think a heartbeat check every few seconds should suffice and should handle potentially large downtimes.   I will keep an eye out here and see if an exponential strategy like one used in the toolkit would be better applicable in this case as well.

Reply:
Thanks! I updated the code per your suggestion here and our chat conversation